x/market/handler/endblock_test.go

```diff
@@ -0,0 +1,125 @@
+package handler
+
+import (
+	"bytes"
+	"errors"
+	"math/rand"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ovrclk/akash/testutil"
+	"github.com/ovrclk/akash/x/market/types"
+)
+
+var (
+	// Create and seed the generator.
+	// Typically a non-fixed seed should be used, such as time.Now().UnixNano().
+	// Using a fixed seed will produce the same output on every run.
+	r = rand.New(rand.NewSource(550000))
+)
+
+type winnerTest struct {
+	desc      string
+	bids      []types.Bid
+	expWinner *types.Bid
+	expErr    error
+}
+
+func (w *winnerTest) testFunc(t *testing.T) {
+	winner, err := pickBidWinner(w.bids)
+	if !errors.Is(err, w.expErr) {
+		t.Errorf("returned err: %v does not match %v", err, w.expErr)
+	}
+	if w.expWinner != nil && !winner.Equals(w.expWinner.ID()) {
+		t.Errorf("unexpected winner: %#v\n%q : %v", winner, types.BidIDString(winner.BidID), winner.Price)
+		t.Logf("winner: %+v coin: %s", winner.ID(), winner.Price.String())
+	}
+}
+
+func TestBidWinner(t *testing.T) {
+	bid0 := createBid(t, uint64(15), int64(3))
+	bid1 := createBid(t, uint64(20), int64(5))
+	bid2 := createBid(t, uint64(19), int64(5))
+	bid3 := createBid(t, uint64(20), int64(4))
+	bid4 := createBid(t, uint64(20), int64(5))
+	bid5 := createBid(t, uint64(19), int64(5))
+	bid6 := createBid(t, uint64(18), int64(5))
+	bid7 := createBid(t, uint64(20), int64(8))
+	bid8 := createBid(t, uint64(20), int64(6))
+	bid9 := createBid(t, uint64(20), int64(5))
+
+	var winnerTests = []winnerTest{
+		{
+			desc:      "no bids",
+			bids:      []types.Bid{},
+			expWinner: nil,
+			expErr:    errNoBids,
+		},
+		{
+			desc:      "single bid",
+			bids:      []types.Bid{bid9},
+			expWinner: &bid9,
+		},
+		{
+			desc:      "two bids",
+			bids:      []types.Bid{bid9, bid7},
+			expWinner: &bid9,
+		},
+		{
+			desc:      "two matching bids",
+			bids:      []types.Bid{bid9, bid7, bid4},
+			expWinner: &bid4,
+		},
+		{
+			desc:      "all the same bid values",
+			bids:      []types.Bid{bid9, bid9, bid9, bid9, bid9},
+			expWinner: &bid9,
+		},
+		{
+			bids:      []types.Bid{bid9, bid3},
+			expWinner: &bid3,
+		},
+		{
+			bids:      []types.Bid{bid1, bid0, bid9, bid2, bid3, bid4},
+			expWinner: &bid0,
+		},
+		{
+			bids:      []types.Bid{bid5, bid6, bid7, bid8},
+			expWinner: &bid6,
+		},
+	}
+
+	for _, test := range winnerTests {
+		t.Run(test.desc, test.testFunc)
+	}
+}
+
+func createBid(t *testing.T, block uint64, bid int64) types.Bid {
+	t.Helper()
+	akashDenom := "akash"
+	seed0, seed1 := make([]byte, 32), make([]byte, 32)
+	_, err := r.Read(seed0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = r.Read(seed1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if bytes.Equal(seed0, seed1) {
+		t.Fatalf("random seed bytes the same")
+	}
+	t.Logf("seed data: %q %q", seed0, seed1)
+	return types.Bid{
+		BidID: types.BidID{
+			Owner:    testutil.DeterministicAccAddress(t, seed0),
```

Comment:
You need the `OrderID` to be the same for all bids - `Owner`, `DSeq`, `GSeq`, `OSeq`.  You can use `testutil.AccAddress(t)` for that and pass it into this function along with `block`.

This constant seed thing is really unnecessary and bloats the codebase.